File: java/Problem7.java
class Problem7 {
    static boolean isPrimeNumber(int n) {
        int mod = 0;
        int divisor = 1;
        int flag = 0;
        while(divisor <= n) {
            mod = n % divisor;
            if(mod == 0 && divisor > 1 && divisor < n)
                flag++;
            divisor++;
        }
        if(flag == 0)
            return true;
        else
            return false;
    }
    static int xPrimeNumber(int x) {
        int lastPrime = 0;
        int i = 2;
        int primeCounter = 0;
        while(primeCounter != x) {
            if(isPrimeNumber(i)) {
                lastPrime = i;
                primeCounter++;
            }
            i++;
        }
        return lastPrime;
    }
    public static void main(String[] args) {
        System.out.println(xPrimeNumber(10001));
    }
}

File: ruby/problem7.rb
def is_prime(n)
    divisor = 1
    div = 0
    counter = 0
    while divisor <= n
        div = n % divisor
        if div == 0 && divisor > 1 && divisor < n
            counter = counter + 1
        end
        divisor = divisor + 1
    end
    if counter == 0
        return true
    else
        return false
    end
end
def x_number_prime(x)
    last_prime = 0
    prime_counter = 0
    i = 2
    while prime_counter != x
        if is_prime(i)
            prime_counter = prime_counter + 1
            last_prime = i
        end
        i = i + 1
    end
    return last_prime
end
puts x_number_prime(10001)
